application/src/main/java/net/corda/v5/application/persistence/PagedQuery.java

```diff
@@ -49,5 +49,33 @@
      */
     @Suspendable
     @NotNull
-    List<R> execute();
+    ResultSet<R> execute();
+
+    /**
+     * A representation of a result set returned by a paged query when executed.
+     *
+     * @param <T> Type of the results contained in the result set.
+     */
+    interface ResultSet<T> {
+
+        /**
+         * @return The results contained in the result set.
+         */
+        List<T> getResults();
+
+        /**
+         * @return The size of the result set.
+         */
+        Integer getSize();
```

Comment:
Should be `int`? Same for the other ones as well.

Reply:
good point